src/main/java/pl/put/poznan/transformer/rest/JSONToolsController.java

```diff
@@ -5,96 +5,132 @@
+import pl.put.poznan.transformer.logic.tools.JSONToolWhitelist;
 
 import java.util.List;
 import java.util.Map;
 
+import static pl.put.poznan.transformer.logic.tools.JSONToolBox.*;
 
 @RestController
 @CrossOrigin
 @RequestMapping("/api/v1/")
 public class JSONToolsController {
     private static final Logger logger = LoggerFactory.getLogger(JSONToolsController.class);
-    private final ObjectMapper objectMapper = new ObjectMapper();
-    private final JSONToolsService service = new JSONToolsService();
 
     @RequestMapping(value="minify", method = { RequestMethod.POST }, produces = "application/json")
     public ResponseEntity<String> minify(
         @RequestBody String payload
     ) {
+        final JSONToolMinify tool = new JSONToolMinify();
+
         logger.info("POST /api/v1/minify");
-        JsonNode jsonNode = service.parse(payload);
-        if(jsonNode == null){
+        JsonNode jsonNode;
+
+        try{
+            jsonNode = parse(payload);
+        }catch(JSONException exception){
             return ResponseEntity.status(400).body("{\"error\":\"Payload is not valid JSON!\"}");
-        }else{
-            JsonNode json = service.getInput(jsonNode);
-            if(json == null){
-                return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
-            }
-            if(jsonNode.size() != 1){
-                return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
-            }
-            try{
-                return ResponseEntity.status(200).body(service.minify(json));
-            }catch(Exception exception){
-                logger.error(exception.toString());
-                return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
-            }
+        }
+
+        if(jsonNode.size() != 1){
+            return ResponseEntity.status(400).body("{\"error\":\"Request body should contain only 'json' property!\"}");
+        }
+
+        JsonNode json;
+        try{
+            json = getInput(jsonNode);
+        }catch(JSONException exception){
+            return ResponseEntity.status(400).body("{\"error\":\"Missing or broken 'json' property!\"}");
+        }
+
+        try{
+            return ResponseEntity.status(200).body(tool.decorate(new JSONObject(json.toString())).getJson());
+        }catch(Exception exception){
+            logger.error(exception.toString());
+            return ResponseEntity.status(500).body("{\"error\":\"Internal error!\"}");
         }
     }
 
     @RequestMapping(value="pretty", method = { RequestMethod.POST }, produces = "application/json")
     public ResponseEntity<String> pretty(
             @RequestBody String payload
     ) {
+        final JSONToolPrettify tool = new JSONToolPrettify();
+
         logger.info("POST /api/v1/pretty");
-        JsonNode jsonNode = service.parse(payload);
-        if(jsonNode == null){
+        JsonNode jsonNode;
```

Comment:
jeżeli dobrze rozumiem to jest wejście i to nie może być JsonNode bo potem dostajemy coś takiego:

2022-12-10 09:58:30.701  INFO 17820 --- [nio-8080-exec-2] p.p.p.t.rest.JSONToolsController         : POST /api/v1/minify
2022-12-10 09:58:30.702 DEBUG 17820 --- [nio-8080-exec-2] p.p.p.t.logic.tools.JSONToolMinify       : Decorating!
2022-12-10 09:58:30.702 DEBUG 17820 --- [nio-8080-exec-2] p.p.p.t.logic.tools.JSONToolMinify       : I am a Starting point
2022-12-10 09:58:30.702  INFO 17820 --- [nio-8080-exec-2] p.p.p.t.logic.tools.JSONToolMinify       : Minifing:
{"example":3}
2022-12-10 09:58:30.702  INFO 17820 --- [nio-8080-exec-2] p.p.p.t.logic.tools.JSONToolMinify       : Minifacion returned:
{"example":3}

Reply:
zmieńto prosze na stringa albo na JSONObject(), bo teraz wejście jest parsowane przez JsonNode zanim jeszcze trafi do naszego parsowania